src/test/functional/tests/buzzPage.spec.ts

```diff
@@ -20,21 +30,68 @@ test.describe("Buzz Page", () => {
 
   test.afterEach(async ({ page }) => {
     buzzPage = new BuzzPage(page);
-    await buzzPage.deleteTheNewestPost()
+    await buzzPage.deleteTheNewestPost(false)
   });
 
   test("Post should be shared", async ({ page }) => {
+    const randomTitle = generateRandomString()
     await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
-    await buzzPage.sharePost(filePath);
+    await buzzPage.sharePost(filePath, randomTitle);
     await page.reload();
-    await expect(buzzPage.photoImg.first()).toBeAttached();
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(randomTitle);
   });
 
   test("Post should be edited", async ({ page }) => {
+    const randomTitle = generateRandomString()
     await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
-    await buzzPage.sharePost(filePath);
+    await buzzPage.sharePost(filePath, randomTitle);
     await page.reload();
     await buzzPage.editTheNewestPost(expectedPostTextAfterEdition)
-    await expect(page.getByText(expectedPostTextAfterEdition)).toBeVisible();
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(expectedPostTextAfterEdition);
+  });
+
+  test("Video should be shared", async ({ page }) => {
+    await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
+    await buzzPage.shareVideo(videoTitle, videoUrl);
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(videoTitle);
+  });
+});
+
+test.describe("Most liked and most commented post", () => {
+  let loginPage: LoginPage;
+  let buzzPage: BuzzPage;
+
+  test.beforeEach(async ({ page }) => {
+    loginPage = new LoginPage(page);
+    buzzPage = new BuzzPage(page);
+    await loginPage.initialize();
+    await loginPage.navigateToMainPage();
+    await loginPage.loginUser(adminUserTestData.userName, adminUserTestData.password);
+    await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
+    await buzzPage.shareVideo(videoTitle, videoUrl);
+    await buzzPage.shareVideo(mostLiked, videoUrl2);
+    await buzzPage.shareVideo(mostCommented, videoUrl3);
+    
+  });
+
+  test.afterEach(async ({ page }) => {
+    buzzPage = new BuzzPage(page);
+    await buzzPage.deleteTheNewestPost(true)
+    await buzzPage.deleteTheNewestPost(true)
+    await buzzPage.deleteTheNewestPost(true)
+  });
+
+  test("Most Liked", async ({ page }) => {
+    await buzzPage.heart.nth(1).click()
+    await buzzPage.mostLikedTab.click()
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(mostLiked)
+  });
+
+  test("Most Commented", async ({ page }) => {
+    await buzzPage.commentPost.nth(0).click()
+    await buzzPage.commentInput.nth(0).type(newComment)
+    await page.keyboard.press('Enter');
+    await buzzPage.mostCommentedTab.click()
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(mostCommented)
```

Comment:
the same as above - why do you check If body text area has 'mostCommented' text? I think you should check comment section, not the body of that post

Reply:
I commented on the post then I changed the tab for most commented post and that post should be in first place in that tab because it only has comments put down so it is most commented do you get it now?

Reply:
Yes, I see what you do. But what If some test failed before and there are more posts there?

Reply:
maybe it would be better to generate random string and check it?